### PR TITLE
Seafile Server Monitoring

### PR DESCRIPTION
**Pull request by SourceDoctor**
Monitoring for a Seafile Server

[About Seafile Server](https://www.seafile.com/en/home/)

### CODE REVIEW

File: includes/html/functions.inc.php [hunk 1]
```diff
@@ -1449,6 +1449,34 @@ function get_arrays_with_mdadm($device, $app_id)
...
+/**
+ * Get all seafile data from the collected
+ * rrd files.
+ *
+ * @param array $device device for which we get the rrd's
+ * @param int   $app_id application id on the device
+ * @param string  $category which category of seafile graphs are searched
+ * @return array list of seafile data
+ */
+function get_arrays_with_seafile($device, $app_id, $category)
```
Comment (murrant) [inline includes/html/functions.inc.php]: this function is almost a duplicate of get_arrays_with_mdadm(),  Can  you combine them?

Comment (SourceDoctor) [reply includes/html/functions.inc.php]: @murrant 
globalized function
which handles now mdadm. smart and seafile

Comment (SourceDoctor) [reply includes/html/functions.inc.php]: @murrant 
i combined it in a globalized function

**Comment by murrant** [comment]
Thanks, looks good, just need test data.  Check the other apps for examples. (linux_fail2ban-v1.json for example)

**Comment by SourceDoctor** [comment]
> Thanks, looks good, just need test data. Check the other apps for examples. (linux_fail2ban-v1.json for example)

i took a look into linux_fail2ban-v1.json but i'm a little bit confused.
Most RRDs on Seafile will be filled with multple value. e.G. Space

**Comment by SourceDoctor** [comment]
@murrant 
ok, i recorded seafile data and saved it in tests/snmpsim/linux_seafile-v1.snmprec.
Then i call in the scripts directory:
su librenms -c "./save-test-data.php -o linux_seafile --snmpsim -d"

but the script seems to hang on:

> ...
>   Directory "/usr/local/lib/python3.5/dist-packages/snmpsim/data" does not exist
>   --- SNMPv3 USM configuration
>   SNMPv3 USM SecurityName: simulator
>   SNMPv3 USM authentication key: auctoritas, authentication protocol: MD5
>   SNMPv3 USM encryption (privacy) key: privatus, encryption protocol: DES
>   Maximum number of variable bindings in SNMP response: 64
>   --- Transport configuration
>   Listening at UDP/IPv4 endpoint 127.1.6.1:1161, transport ID 1.3.6.1.6.1.1.0
> 
some ideas?

**Comment by SourceDoctor** [comment]
@murrant
ok, i recorded seafile data and saved it in tests/snmpsim/linux_seafile-v1.snmprec.
Then i call in the scripts directory:
su librenms -c "./save-test-data.php -o linux_seafile --snmpsim -d"

./scripts/pre-commit.php --db --snmpsim
runs without problems

> .....
> Checked 3479 files in 29.1 seconds
> No syntax error found
> 

but json didn't seems to be recorded

**Comment by SourceDoctor** [comment]
@murrant 
thx for the json. But i'm confused about the travis result.
php 7.2 run's well but php 7.1 and php 7.3 are failing.
i tested the json file and it is in correct format.

**Comment by laf** [comment]
@SourceDoctor I _think_ it's because it needs to be hex encoded. See if that edit works.

**Comment by laf** [comment]
Thanks for contributing @SourceDoctor :)

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-57-release-changelog-october-2019/9969/1

**Pull request merged**